### PR TITLE
Patch to increase test coverage

### PR DESCRIPTION
**Pull request by icyblade**
This PR will try to increase test coverage of Keras under Python 3.6 and TensorFlow backend, including:

- Add unit test for `keras.callbacks.RemoteMonitor`
- Enable unit test for `keras.utils.vis_utils` under Python 3.6.
BTW, current test seems working in my environment. I wonder why it has been disabled before.
- Add unit test for `keras.preprocessing.sequence._remove_long_seq`
- Add unit test for `keras.utils.io_utils.HDF5Matrix`, related to `__getitem__` and `normalizer`
- Add unit test for `keras.utils.io_utils.ask_to_proceed_with_overwrite`

### CODE REVIEW

File: tests/keras/test_callbacks.py [hunk 1]
```diff
@@ -787,5 +787,29 @@ def test_TensorBoard_with_ReduceLROnPlateau(tmpdir):
...
+    y_train = np_utils.to_categorical(y_train)
+    model = Sequential()
+    model.add(Dense(num_hidden, input_dim=input_dim, activation='relu'))
+    model.add(Dense(num_classes, activation='softmax'))
+    model.compile(loss='categorical_crossentropy',
+                  optimizer='rmsprop',
+                  metrics=['accuracy'])
+    mode = 'max'
+    monitor = 'val_acc'
+    patience = 0
```
Comment (fchollet) [inline tests/keras/test_callbacks.py]: These quantities are not relevant here.

File: tests/keras/test_callbacks.py [hunk 2]
```diff
@@ -787,5 +787,29 @@ def test_TensorBoard_with_ReduceLROnPlateau(tmpdir):
...
+    cbks = [callbacks.RemoteMonitor()]
+    history = model.fit(X_train, y_train, batch_size=batch_size,
+                        validation_data=(X_test, y_test), callbacks=cbks, epochs=20)
```
Comment (fchollet) [inline tests/keras/test_callbacks.py]: You only need one epoch. Also, this line will simply fail at the following try/except block and raise a warning:

```python
        try:
            requests.post(self.root + self.path,
                          {self.field: json.dumps(send)},
                          headers=self.headers)
        except requests.exceptions.RequestException:
            warnings.warn('Warning: could not reach RemoteMonitor '
                          'root server at ' + str(self.root))
```

File: tests/keras/test_callbacks.py [hunk 3]
```diff
@@ -787,5 +787,29 @@ def test_TensorBoard_with_ReduceLROnPlateau(tmpdir):
     assert not tmpdir.listdir()
 
 
+@keras_test
+def tests_RemoteMonitor():
+    np.random.seed(1337)
```
Comment (fchollet) [inline tests/keras/test_callbacks.py]: You don't need to seed a test that doesn't do value-based correctness testing.

File: tests/keras/utils/vis_utils_test.py [hunk 4]
```diff
@@ -11,7 +11,6 @@
 from keras.utils import vis_utils
 
 
-@pytest.mark.skipif(sys.version_info > (3, 0), reason='pydot-ng currently supports python 3.4')
```
Comment (taehoonlee) [inline tests/keras/utils/vis_utils_test.py]: The tests are failing. Would you please add `pip install pydot graphviz;` to the line 71 on `.travis.yml`?

File: tests/keras/test_callbacks.py (hunk 2 above)
Comment (icyblade) [reply tests/keras/test_callbacks.py]: Should I set up a HTTP server to test this module, or just monkey patch `requests.post`?

**Comment by icyblade** [comment]
@fchollet is this PR ready to be merged?
